### PR TITLE
feat: article on exporting PDFs

### PR DESCRIPTION
**Pull request by philsturgeon**

### CODE REVIEW

**Comment by philsturgeon** [comment]
@mnaumanali94 sorry for flinging a few reviews at you today and yesterday. The last few articles you've done have looked amazing thanks to your graphics, could you help me out again? Also another pair of eyes on the content here would be really appreciated. This was rather boring to write and my brain has slightly turned off, so there's probably some glaring mistakes, but this will get us a bunch of SEO I think.

**Comment by philsturgeon** [comment]
@lottamus you were asking me for OpenAPI to PDF guidance so I wrote this. If you can help me review it, and fire it off to whomever was asking about it, that would be amazing. I cannot think about PDFs anymore please! :)

**Comment by lottamus** [comment]
:heavy_check_mark: Deploy preview for *stoplightio* ready!


:hammer: Explore the source changes: 52b8bd33e42dae26478e7921032edbb7e2856d27

:mag: Inspect the deploy logs: [https://app.netlify.com/sites/stoplightio/deploys/5ff36eede4990e0008810411](https://app.netlify.com/sites/stoplightio/deploys/5ff36eede4990e0008810411?utm_source=github&utm_campaign=bot_dl)

:sunglasses: Browse the preview: [https://deploy-preview-1706--stoplightio.netlify.app](https://deploy-preview-1706--stoplightio.netlify.app?utm_source=github&utm_campaign=bot_dp)

**Comment by lottamus** [comment]
:heavy_check_mark: Deploy preview for *stoplightio* ready!


:hammer: Explore the source changes: 4fd889f3031c614452b7e561f9731c39948c70d9

:mag: Inspect the deploy logs: [https://app.netlify.com/sites/stoplightio/deploys/5ff370136338470007e35941](https://app.netlify.com/sites/stoplightio/deploys/5ff370136338470007e35941?utm_source=github&utm_campaign=bot_dl)

:sunglasses: Browse the preview: [https://deploy-preview-1706--stoplightio.netlify.app](https://deploy-preview-1706--stoplightio.netlify.app?utm_source=github&utm_campaign=bot_dp)

**Comment by mnaumanali94** [comment]
For sure @philsturgeon. I'll get to this a bit later in the day

File: netlify/blog-posts/pdf-api-documentation.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  twitter:
+    description: >-
+      Using OpenAPI to create PDF based reference documentation for clients and other third-parties
+    title: >-
+      Export API Documentation as a PDF
+    image: /images/blog-images/openapi-to-pdf.png
+    url: https://stoplight.io/blog/pdf-api-documentation/
+---
+
+You've built a brilliant API, well done! But hold up, people are trying to use it, and there's no public facing documentation. Your API and the docs are private, and require a special VPN, SAML, and/or some sort of OAuth proxy that requires not just a GitHub account, but that GitHub user needs to be part of your organization.
```
Comment (mnaumanali94) [inline netlify/blog-posts/pdf-api-documentation.md:42]:
```suggestion
You've built a brilliant API, well done! But hold up, people are trying to use it, and there's no public-facing documentation. Your API and the docs are private, and require a special VPN, SAML, and/or some sort of OAuth proxy that requires not just a GitHub account, but that GitHub user needs to be part of your organization.
```

File: netlify/blog-posts/pdf-api-documentation.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+Whatever the approach, there are often users outside of this super-secure realm that need access to the documentation. Maybe it's a customer who is paying a bunch to get access to some of your data. Maybe its a contractor who is going to be doing some work on the API and doesn't want to [guess the contract](https://apisyouwonthate.com/blog/commit-to-api-contracts). Maybe it's a different department who are in a different GitHub organization, e.g.: an acquired company wants to integrate.
```
Comment (mnaumanali94) [inline netlify/blog-posts/pdf-api-documentation.md:44]:
```suggestion
Whatever the approach, there are often users outside of this super-secure realm that needs access to the documentation. Maybe it's a customer who is paying a bunch to get access to some of your data. Maybe it's a contractor who is going to be doing some work on the API and doesn't want to [guess the contract](https://apisyouwonthate.com/blog/commit-to-api-contracts). Maybe it's a different department that is in a different GitHub organization, e.g.: an acquired company wants to integrate.
```

File: netlify/blog-posts/pdf-api-documentation.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+![Screenshot of a POST request in the PDF produced by swagger2pdf](/images/blog/pdf-api-documentation/rapipdf.png)
+
+As this one is a bit tricky to show off in a single screenshot, take a look at [the resulting PDF here](/images/blog/pdf-api-documentation/rapipdf-instagram-example.pdf).
+
+## Alternative Workflows
+
+Instead of doing any of this, it might be worth taking a step back, and thinking "Why do I need to make a PDF?" Somebody may have asked you to do it, or you've decided to do it because it's something you've always done, but... do you need to?
+
+The base use-case here is often an API that may or many not be private, but definitely has private or "internal" documentation, and somebody who is not in that "in group" needs access to it. There are a lot of other ways to do this.
```
Comment (mnaumanali94) [inline netlify/blog-posts/pdf-api-documentation.md:122]:
```suggestion
The base use-case here is often an API that may or may not be private, but definitely has private or "internal" documentation, and somebody who is not in that "in group" needs access to it. There are a lot of other ways to do this. 
```

File: netlify/blog-posts/pdf-api-documentation.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+The base use-case here is often an API that may or many not be private, but definitely has private or "internal" documentation, and somebody who is not in that "in group" needs access to it. There are a lot of other ways to do this. 
+
+For starters, you could generate HTML with any of the many free and open-source tools that create [Beautiful API Reference Documentation](https://apisyouwonthate.com/blog/turning-contracts-into-beautiful-documentation), and host that on a [password-protected Amazon S3 bucket](https://kynatro.com/blog/2018/01/03/a-step-by-step-guide-to-creating-a-password-protected-s3-bucket/) then email the password to and a URL instead of emailing a PDF. If the S3 bucket is updated with new HTML by your continuous integration server then you don't have to worry about emailing over updated PDFs.
+
+If you don't want to set up _any_ of this, Stoplight handles the workflow for you out of thr box. There's no need to email PDFs around, or roll your own password-protected hosted documentation. We already have private hosted documentation based on SAML, and version-control-based SSO from GitHub, BitBucket, etc., then you can whitelist email address domains too.
```
Comment (mnaumanali94) [inline netlify/blog-posts/pdf-api-documentation.md:126]:
```suggestion
If you don't want to set up _any_ of this, Stoplight handles the workflow for you out of the box. There's no need to email PDFs around, or roll your own password-protected hosted documentation. We already have private hosted documentation based on SAML, and version-control-based SSO from GitHub, BitBucket, etc., then you can whitelist email address domains too. 
```

File: netlify/blog-posts/pdf-api-documentation.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+If you don't want to set up _any_ of this, Stoplight handles the workflow for you out of thr box. There's no need to email PDFs around, or roll your own password-protected hosted documentation. We already have private hosted documentation based on SAML, and version-control-based SSO from GitHub, BitBucket, etc., then you can whitelist email address domains too. 
+
+Maybe you work at `protectedcompany.com` and you want anyone at `somecustomer.com` to get access to certain projects? Whitelist `somecustomer.com` with Guest priviledges, get them to sign up, and add them to the specific projects they should be allowed to see. Maybe you have a contractor who should be treated like an employee and have access to seeing most APIs? Make them a Viewer and now they can see any internal projects.
```
Comment (mnaumanali94) [inline netlify/blog-posts/pdf-api-documentation.md:128]:
```suggestion
Maybe you work at `protectedcompany.com` and you want anyone at `somecustomer.com` to get access to certain projects? Whitelist `somecustomer.com` with Guest privileges, get them to sign-up, and add them to the specific projects they should be allowed to see. Maybe you have a contractor who should be treated like an employee and have access to seeing most APIs? Make them a Viewer and now they can see any internal projects.
```

File: netlify/blog-posts/pdf-api-documentation.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+
+So what do you do? A common approach is to export the API docs as a PDF, then email it over. Sure. That'll work, but _how_?!
+
+## Step 1: Create an OpenAPI Description
+
+For a lot of you reading this, you'll already have an OpenAPI description. Someone on your team probably [created it before anyone started programming the API](https://stoplight.io/api-design-guide/basics/#why-api-design-first-matters), so that everyone could be sure the right thing was being built in the first place. Other teams might have [generated OpenAPI using annotations](https://stoplight.io/blog/generate-documentation-from-code/) added to the code.
+
+If you've not got any OpenAPI yet head over to [Stoplight Studio](/studio) or look for another [Visual OpenAPI Editor](https://openapi.tools/#gui-editors) to put together some OpenAPI for your API. Each PDF conversion approach will have an example you can try.
+
+## Step 2: Convert OpenAPI to PDF
```
Comment (mnaumanali94) [inline netlify/blog-posts/pdf-api-documentation.md:54]:
```suggestion
## Step 2: Convert to PDF
```

File: netlify/blog-posts/pdf-api-documentation.md [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+
+The popular API documentation tools use the "three-column" layout, popularized by [Stripe API Reference](https://stripe.com/docs/api) with a navigation bar on the left and other stuff on the right, which looks daft squashed onto an A4 page.
+
+Then there's the interactive elements in most good API documentation, like interactive HTTP consoles, parameter lists with expandable nested content, and generally a bunch of hyperlinks that don't work when you print them. 
+
+![Stripe often considered the best API docs around, look rather awkward and unusable when printed due to collapsed nested objects not being expandable.](/images/blog/pdf-api-documentation/print-comparison.jpg)
+
+If your API docs are using a really clever `print.css` that hides everything not useful in a print context, and expanded everything that should be expanded, it might just work out fine. This is rare, but worth trying. If it doesn't work, try another one of these approaches.
+
+### swagger2pdf
```
Comment (mnaumanali94) [inline netlify/blog-posts/pdf-api-documentation.md:68]:
```suggestion
### Swagger2pdf
```

Looks weird with other headings

**Review by mnaumanali94 — APPROVED** [review]
Also check out the cover i just added. Apologies for the delay, i was out yesterday

**Comment by philsturgeon** [comment]
That's not a delay thats bloody quick, thank you for the graphic and the suggestions @mnaumanali94 !

**Pull request merged**